Automattic-Tracks-iOSTests/Mocks.swift

```diff
@@ -0,0 +1,22 @@
+import Foundation
+import Sodium
+@testable import AutomatticTracks
+
+typealias LogFileCallback = (LogFile) -> ()
+typealias ErrorWithLogFileCallback = (Error, LogFile) -> ()
+
+enum MockError: Error {
+    case generic
+}
+
+class MockLogFile: LogFile {
+    init(string: String) {
+        let uuid = UUID().uuidString
+        let logFileURL = FileManager.default.createTempFile(named: uuid, containing: string)
+        super.init(url: logFileURL, uuid: uuid)
+    }
+
+    static var withRandomString: LogFile {
+        return MockLogFile(string: String.randomString(length: 128))
+    }
+}
```

Comment:
I'm good with this, and it's implemented as part of fcb11cb509139e33e0a4fda70292f16c7925542b.

Instead of a utility function, I added an extension for `LogFile`, and implemented this as a convenience initializer for `LogFile`. Lastly, because in most cases we don't actually care about `self.log`, I simplified the test class to remove it, and brought the `LogFile.containingRandomString` forward from a future PR as a substitute.